src/Maker/MakeResetPassword.php

```diff
@@ -194,23 +203,84 @@ public function generate(InputInterface $input, ConsoleStyle $io, Generator $gen
+
+        $generator->writeChanges();
+
+        $manipulator = new ClassSourceManipulator(
+            $this->fileManager->getFileContents($pathRequestEntity)
         );
 
-        $generator->generateClass(
-            $repositoryClassNameDetails->getFullName(),
-            'resetPassword/ResetPasswordRequestRepository.tpl.php',
-            [
-                'request_class_full_name' => $requestClassNameDetails->getFullName(),
-                'request_class_name' => $requestClassNameDetails->getShortName(),
-            ]
+        $manipulator->addInterface(ResetPasswordRequestInterface::class);
+
+        $manipulator->clearClassNodeStmts();
+
+        $manipulator->addTrait(ResetPasswordRequestTrait::class);
+
+        $manipulator->addConstructor([
+                ['user', null, 'object'],
+                ['expiresAt', null, '\DateTimeInterface'],
+                ['selector', null, 'string'],
+                ['hashedToken', null, 'string'],
+            ], <<<'CODE'
+<?php
+$this->user = $user;
+$this->initialize($expiresAt, $selector, $hashedToken);
+CODE
+        );
+        $manipulator->addProperty('id', [
+            '@ORM\Id()',
+            '@ORM\GeneratedValue()',
+            '@ORM\Column(type="integer")',
+        ]);
+
+        $manipulator->addProperty('user', [
+            '@ORM\ManyToOne(targetEntity="App\Entity\User")',
+        ]);
+
+        $manipulator->addGetter('user', 'object', false);
+
+        $this->fileManager->dumpFile(
+            $pathRequestEntity,
+            $manipulator->getSourceCode()
+        );
+
+        $pathRequestRepository = $this->fileManager->getRelativePathForFutureClass(
+            $repositoryClassNameDetails->getFullName()
+        );
+
+        $manipulator = new ClassSourceManipulator(
+            $this->fileManager->getFileContents($pathRequestRepository)
+        );
+
+        $manipulator->clearClassNodeStmts();
+
+        $manipulator->addInterface(ResetPasswordRequestRepositoryInterface::class);
+
+        $manipulator->addTrait(ResetPasswordRequestRepositoryTrait::class);
+
+        $manipulator->addConstructor([
+                ['registry', null, 'ManagerRegistry'],
+            ], <<<'CODE'
+<?php
+parent::__construct($registry, ResetPasswordRequest::class);
+CODE
+        );
+        $methodBuilder = $manipulator->createMethodBuilder('createResetPasswordRequest', ResetPasswordRequestInterface::class, false);
+
+        $manipulator->addMethodBuilder($methodBuilder, [
+                ['user', null, 'object'],
+                ['expiresAt', null, '\DateTimeInterface'],
+                ['selector', null, 'string'],
+                ['hashedToken', null, 'string'],
+            ], <<<'CODE'
+<?php
+return new ResetPasswordRequest($user, $expiresAt, $selector, $hashedToken);
+CODE
         );
+        $this->fileManager->dumpFile($pathRequestRepository, $manipulator->getSourceCode());
```

Comment:
I think putting all of this new code into a private method in this class might be nice. Or, its own class, which could then be unit tested a bit easier - like `UserClassBuilder`

Reply:
I don't understand, all the code or a specific part ? You just select one line

Reply:
When we are okay for this issue https://github.com/symfony/maker-bundle/pull/590#discussion_r422145767
I start to think about.

Reply:
I added a new private like you said in first because i think that this stuff is really specific to this command, not really a UserClass wich is built. See if you like :)